src/features/emails/rpc/copyEmail.ts

```diff
@@ -0,0 +1,52 @@
+import { z } from 'zod';
+
+import IApiClient from 'core/api/client/IApiClient';
+import { makeRPCDef } from 'core/rpc/types';
+import {
+  ZetkinEmail,
+  ZetkinEmailPostBody,
+  ZetkinQuery,
+} from 'utils/types/zetkin';
+
+const paramsSchema = z.object({
+  emailId: z.number(),
+  orgId: z.number(),
+});
+type Params = z.input<typeof paramsSchema>;
+type Result = ZetkinEmail;
+
+export const copyEmailDef = {
+  handler: handle,
+  name: 'copyEmail',
+  schema: paramsSchema,
+};
+
+export default makeRPCDef<Params, Result>(copyEmailDef.name);
+
+async function handle(params: Params, apiClient: IApiClient) {
+  const { emailId, orgId } = params;
+
+  const email = await apiClient.get<ZetkinEmail>(
+    `/api/orgs/${orgId}/emails/${emailId}`
+  );
+
+  const createdEmail = await apiClient.post<ZetkinEmail, ZetkinEmailPostBody>(
+    `/api/orgs/${orgId}/emails`,
+    {
+      campaign_id: email.campaign.id,
+      content: email.content,
+      subject: email.subject,
+      title: email.title,
+    }
+  );
+  await apiClient.patch<ZetkinQuery, Partial<ZetkinQuery>>(
+    `/api/orgs/${orgId}/people/queries/${createdEmail.target.id}`,
+    { filter_spec: email.target.filter_spec }
+  );
+
+  const copiedEmail = await apiClient.get<ZetkinEmail>(
+    `/api/orgs/${orgId}/emails/${createdEmail.id}`
+  );
```

Comment:
Why do you need to retrieve it again? I worry that it will take an unnecessary amount of time because another API request is made. Can you maybe just return `createdEmail`? If the reason is that you want to update the query, maybe do:

```tsx
return {
  ...copiedEmail,
  target: {
    ...copiedEmail.target,
    filter_spec: email.target.filter_spec,
  }
};
```